File: scratch/test_api.py
import urllib.request
import json

base_url = 'http://192.168.1.25:8000/api'

# Login
login_data = json.dumps({
    "email": "cs@klinklin.com",
    "pin": "123456"
}).encode('utf-8')

req = urllib.request.Request(f"{base_url}/login", data=login_data, headers={'Content-Type': 'application/json', 'Accept': 'application/json'})

try:
    with urllib.request.urlopen(req) as res:
        login_res = json.loads(res.read().decode('utf-8'))
        token = login_res.get('token')
        print(f"Token: {token}")

        # Fetch Karyawans
        req2 = urllib.request.Request(f"{base_url}/karyawans", headers={'Authorization': f'Bearer {token}', 'Accept': 'application/json'})
        with urllib.request.urlopen(req2) as res2:
            data = json.loads(res2.read().decode('utf-8'))
            items = data.get('data', [])
            if isinstance(items, dict) and 'data' in items:
                items = items['data']
            
            if items:
                print("First Karyawan:")
                print(json.dumps(items[0], indent=2))
            else:
                print(f"Data: {data}")
except urllib.error.HTTPError as e:
    print(f"HTTP Error: {e.code}")
    print(e.read().decode('utf-8'))
except Exception as e:
    print(f"Error: {e}")

File: scratch/test_api.dart
import 'package:dio/dio.dart';

void main() async {
  final dio = Dio(BaseOptions(baseUrl: 'http://192.168.1.25:8000/api'));
  try {
    print('Logging in...');
    final res = await dio.post('/login', data: {
      "email": "cs@klinklin.com",
      "pin": "123456"
    });
    final token = res.data['token'];
    print('Token: $token');
    
    print('Fetching /karyawans...');
    final res2 = await dio.get('/karyawans', options: Options(headers: {'Authorization': 'Bearer $token'}));
    
    final data = res2.data;
    if (data['data'] != null && data['data'] is List && (data['data'] as List).isNotEmpty) {
      print('First Karyawan:');
      print(data['data'].first);
    } else {
      print('Data: $data');
    }
  } catch (e) {
    print('Error: $e');
    if (e is DioException) {
      print('Response: ${e.response?.data}');
    }
  }
}
